Upgrade Code to Python 3

omuzzyEBSCO opened the pull request:
Updating to Python 3 - do not merge (work in progress).

omuzzyEBSCO committed:
First pass

omuzzyEBSCO committed:
Second pass

jjensen-ebsco commented on ebscopy/ebscopy.py:
```diff
@@ -50,13 +50,14 @@ def _parse_highlight(text):
 
 	start_tag								= "&lt;highlight&gt;"
-	end_tag									= "&lt;\/highlight&gt;"
+	end_tag									= "&lt;/highlight&gt;"
```
Is the linting suggesting this?

omuzzyEBSCO replied:
Yes - with the backslash as it was, pylint gave this error: Anomalous backslash in string: '\/'. String constant might be missing an r prefix.

jjensen-ebsco replied:
Might have to define this more intelligently. I assume I did this so I could use it multiple times later on.

jjensen-ebsco commented on ebscopy/ebscopy.py:
```diff
@@ -240,7 +242,7 @@ def _change_eds_date_limiter_to_api(l):
 	# 19780515-1	(bad end)
 	#
-	if l["Id"] != "DT1" or not re.search("\A\s*(?=-?\d{8})(?:\d{8})?-(?:\d{8})?\Z", l["Values"][0]):
+	if l["Id"] != "DT1" or not re.search("\\A\\s*(?=-?\\d{8})(?:\\d{8})?-(?:\\d{8})?\\Z", l["Values"][0]):
```
Is this a fix, a change due to version, or just a linter preference?

omuzzyEBSCO replied:
Linting picked this up - adding the additional backslash cleared the "Anomalous backslash in string: '\A'. String constant might be missing an r prefix." warning.

omuzzyEBSCO replied:
I think I need to review this concept - I think the additional backslash is escaping the character entirely.

jjensen-ebsco replied:
Yeah, I don't trust the linter on complex regular expressions. We might have to simplify the regular expression or definite it more cleanly.

jjensen-ebsco commented on ebscopy/ebscopy.py:
```diff
@@ -376,15 +375,15 @@ def request(self, method, data, session_token=None, attempt=0):
-			if method is not "UIDAuth":
-				raise ValueError("Missing Authentication Token!")
+		except ValueError as e:
+			if method != "UIDAuth":
```
Is this correct?
https://realpython.com/python-is-identity-vs-equality/

omuzzyEBSCO replied:
No, it appears not :)

omuzzyEBSCO committed:
Update ebscopy.py

omuzzyEBSCO committed:
Draft - update to ebscopy

omuzzyEBSCO committed:
Removing comments, renaming date variable to parsed_date and added back != for line 381

omuzzyEBSCO committed:
Creation of TOML file and function to parse version

omuzzyEBSCO committed:
Moved toml_parse() function to top of Utility functions, removed importlib.metadata and _version variable at end of file

omuzzyEBSCO committed:
Made create_session function within _Connection class public to resolve Pylint error W0212

omuzzyEBSCO committed:
added comments for toml_parse function

omuzzyEBSCO committed:
Delete setup.py

omuzzyEBSCO committed:
added r and backslash back to string

omuzzyEBSCO committed:
renamed _version variable to version

omuzzyEBSCO committed:
changed requires-python version in pyproject.toml

omuzzyEBSCO committed:
converted tabs to spaces to remove Bad Indentation linter error

omuzzyEBSCO committed:
Merge pull request #19 from ebsco/feature/om-upgrade-to-3-formatting

converted tabs to spaces to remove Bad Indentation linter error

omuzzyEBSCO committed:
changed license line in pyproject.toml file

omuzzyEBSCO committed:
change ebscopy.py to edsapi.py, changed import statement for ebscopy package in __init__

omuzzyEBSCO committed:
changed version date (testing purposes)

omuzzyEBSCO committed:
added lxml and python-dateutil as dependencies in pyproject.toml

omuzzyEBSCO committed:
changed import statements on ebscopy_test.py, changed ebscopy. to edsapi., and removed standard Python library modules in pyproject.toml dependencies

omuzzyEBSCO committed:
removed wildcard import of ebscopy.edsapi, unused HTTPError import, and removed unicode and Regexpmatches from unittest (Py 2 > 3 differences)

omuzzyEBSCO committed:
removed [tool.hatch.version] table to support static versioning.

omuzzyEBSCO committed:
Update to edsapi.py after pytest/coverage testing

Big commit covering changes implemented after running pytest, coverage reports.

omuzzyEBSCO committed:
__init__ changes

changes to import statements for ebscopy\__init__.py and ebscopy\tests\__init__.py

omuzzyEBSCO committed:
version update for pyproject.toml

Just a version change for testing to pyproject.toml

omuzzyEBSCO committed:
updates to plain_old_test.py

Changed import statement, changed print statements for Python 3 compatibility

omuzzyEBSCO committed:
ebscopy_test.py > edsapi_test.py

renamed ebscopy_test.py to edsapi_test.py. A few small other changes after the file rename (so they are lost in the red/green (sorry). Import statement on line 10-11, comment out unused variable on line 49, added 's' to http on line 362. All instances of 'ebscopy.' changed to 'edsapi.'

jjensen-ebsco committed:
update gitignore and remove pycache file

jjensen-ebsco committed:
added good unit tests

jjensen-ebsco committed:
minor fixes

jjensen-ebsco committed:
moved TODO items to Issues in GitHub, removed commented out logging code

jjensen-ebsco committed:
fixed issue when ENV variables exist during test

jjensen-ebsco committed:
fixed logging, removed version in code

jjensen-ebsco committed:
add exclude to pyproject (not working?)

jjensen-ebsco committed:
E N V I R O M E N T

jjensen-ebsco committed:
hard coded version for now

omuzzyEBSCO committed:
Update README.md

updated import statement from ebscopy to edsapi

jjensen-ebsco committed:
update docs
